compiler/rustc_hir/src/def.rs

```diff
@@ -630,7 +630,7 @@ impl<Id> Res<Id> {
             | Res::SelfTyParam { .. }
             | Res::SelfTyAlias { .. }
             | Res::SelfCtor(..)
-            | Res::ToolMod
+            | Res::ToolMod { .. }
```

Comment:
I believe that all the `Res::ToolMod { .. }` changes can be reverted to not have the braces (keeping this PR's diff shorter).

Reply:
Fixed in https://github.com/rust-lang/rust/pull/111780/commits/08744dc43d48a2e70369e33af49cd4144a81a599